preprocessing/multi_file_example.py

```diff
@@ -0,0 +1,65 @@
+import pandas as pd
+import numpy as np
+import pathlib
+import os
+from tqdm import tqdm
+
+# one-liner for copying data 
+# for((i=1; i <= 100; i++)); do cp source_iris/iris.data "iris_dataset/iris_${i}.data"; done
+
+source_path = os.getenv("SOURCE_PATH")
+if source_path is None: raise Exception("Missing SOURCE_PATH variable")
+
+pattern = os.getenv("PATTERN")
+if pattern is None: raise Exception("Missing PATTERN variable")
+
+source_path = pathlib.Path(".") / source_path
+data_path = pathlib.Path("datasets") / source_path.stem
+data_path.mkdir(exist_ok=True)
+
+rows = 0
+cols = 0
+for i, file in enumerate(tqdm(list(source_path.glob(pattern)))):
+    df = pd.read_csv(file, header=None)
+    df_rows, df_cols = df.shape
+    rows += df_rows
+    cols = df_cols
+
+train_file = np.memmap(data_path / "train.memmap", dtype='float32', mode='w+', shape=(rows, cols -1))
+target_file = np.memmap(data_path / "target.memmap", dtype='int64', mode='w+', shape=(rows))
```

Comment:
basically have to accept that I'll go through all the files once just to get the shape - then the end user has to be told (1) `dtype` to read in and (2) the shape. 

Just knowing the columns would be enough to satisfy shape (or you could assume `target count == rows`) and fill the rest of the shape with `-1`